test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,1777 @@
+            # Get path to PBS_HOME on this host
+            host_pbs_config = self.du.parse_pbs_config(hostname=host)
+            if "PBS_HOME" not in host_pbs_config.keys():
+                self.logger.info("PBS_HOME not found on host " + host)
+                return
+            pbs_home = host_pbs_config["PBS_HOME"]
+        else:
+            pbs_home = self.server.pbs_conf["PBS_HOME"]
+
+        pbs_logdir = os.path.join(pbs_home, "mom_logs")
+        snap_logdir = os.path.join(self.snapdir, MOM_LOGS_PATH)
+        if host != self.server_host:
+            snap_logdir = os.path.join(snap_logdir, host)
+        self.__capture_logs(host, pbs_logdir, snap_logdir,
+                            self.num_daemon_logs)
+
+    def __capture_comm_logs(self, host):
+        """
+        Capture pbs_comm logs from the host specified
+        """
+        if host != self.server_host:
+            # Get path to PBS_HOME on this host
+            host_pbs_config = self.du.parse_pbs_config(hostname=host)
+            if "PBS_HOME" not in host_pbs_config.keys():
+                self.logger.info("PBS_HOME not found on host " + host)
+                return
+            pbs_home = host_pbs_config["PBS_HOME"]
+        else:
+            pbs_home = self.server.pbs_conf["PBS_HOME"]
+
+            # Capture comm_logs
+            pbs_logdir = os.path.join(pbs_home, "comm_logs")
+            snap_logdir = os.path.join(self.snapdir, COMM_LOGS_PATH)
+            if host != self.server_host:
+                snap_logdir = os.path.join(snap_logdir, host)
+            self.__capture_logs(host, pbs_logdir, snap_logdir,
+                                self.num_daemon_logs)
+
+    def __capture_rscs_all(self):
+        """
+        Capture built-in as well as custom resources in a file called
+        'rscs_all'
+        """
+        # Parse all custom resources
+        resources = self.server.parse_resources()
+        if resources is not None:
+            # Convert type and flags to their numeric format
+            for rsc in resources.values():
+                if rsc.type:
+                    rsc.set_type(self.__convert_type_to_numeric(rsc.type))
+                if rsc.flag:
+                    rsc.set_flag(str(self.__convert_flag_to_numeric(rsc.flag)))
+
+            custom_rscs_names = resources.keys()
+
+            # anonymize custom resources
+            if self.anonymize is True and self.anon_obj.resc_key is not None:
+                self.logger.debug("Anonymizing custom resources")
+                resources = self.anon_obj.anonymize_resource_def(resources)
+                custom_rscs_names = resources.keys()
+
+        # The resourcedef file doesn't define the built-in resources.
+        # Use the build-ins specified in the string "BUILT_IN_RSCS"
+        # This string should be updated with each release PBSPro or
+        # a better way of querying built-ins should be implemented
+        builtin_rscs_names = []
```

Comment:
I think you don't need to process builtin rscs string, you can just write that as it is in to RSCS_PATH